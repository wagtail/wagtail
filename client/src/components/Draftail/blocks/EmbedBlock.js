import PropTypes from 'prop-types';
import React from 'react';

import { gettext } from '../../../utils/gettext';
import MediaBlock from './MediaBlock';
import { shortenLabel } from '../decorators/TooltipEntity';
/**
 * Editor block to display media and edit content.
 */
const EmbedBlock = (props) => {
  const { entity, onEditEntity, onRemoveEntity } = props.blockProps;
  const { url, title, thumbnail, providerName, authorName } = entity.getData();
  // Fallback text is used when there is no image available for the embed.
  // In those cases, it seems like author and provider name will always be present.
  const fallbackText = [authorName, providerName]
    .filter((text) => !!text)
    .map(shortenLabel)
    .join('\n');

  return (
    <MediaBlock {...props} src={thumbnail} alt="" fallbackText={fallbackText}>
      {url ? (
        <a
          className="Tooltip__link EmbedBlock__link"
          href={url}
          title={url}
          target="_blank"
          rel="noreferrer"
        >
          {title}
        </a>
      ) : null}
      <button
        className="button Tooltip__button"
        type="button"
        onClick={onEditEntity}
      >
        {gettext('Edit')}
      </button>
      <button
        className="button button-secondary no Tooltip__button"
        onClick={onRemoveEntity}
      >
        {gettext('Delete')}
      </button>
    </MediaBlock>
  );
};

EmbedBlock.propTypes = {
  blockProps: PropTypes.shape({
    entity: PropTypes.object,
  }).isRequired,
};

export default EmbedBlock;
