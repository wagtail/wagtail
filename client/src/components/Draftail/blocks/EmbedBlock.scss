.EmbedBlock__link {
  display: block;
  margin-bottom: $button-spacing;
}
