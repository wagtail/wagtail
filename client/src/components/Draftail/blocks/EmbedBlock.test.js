import React from 'react';
import { shallow } from 'enzyme';

import EmbedBlock from '../blocks/EmbedBlock';

describe('EmbedBlock', () => {
  it('renders', () => {
    expect(
      shallow(
        <EmbedBlock
          block={{}}
          blockProps={{
            editorState: {},
            entityType: {},
            entity: {
              getData: () => ({
                url: 'http://www.example.com/',
                title: 'Test title',
                thumbnail: 'http://www.example.com/example.png',
              }),
            },
            onChange: () => {},
          }}
        />,
      ),
    ).toMatchSnapshot();
  });

  it('no data', () => {
    expect(
      shallow(
        <EmbedBlock
          block={{}}
          blockProps={{
            editorState: {},
            entityType: {},
            entity: {
              getData: () => ({}),
            },
            onChange: () => {},
          }}
        />,
      ),
    ).toMatchSnapshot();
  });
});
