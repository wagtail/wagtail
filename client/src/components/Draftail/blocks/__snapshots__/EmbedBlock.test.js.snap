// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmbedBlock no data 1`] = `
<MediaBlock
  alt=""
  blockProps={
    Object {
      "entity": Object {
        "getData": [Function],
      },
      "entityType": Object {},
    }
  }
  src={null}
>
  <button
    className="button button-secondary no Tooltip__button"
  >
    Delete
  </button>
</MediaBlock>
`;

exports[`EmbedBlock renders 1`] = `
<MediaBlock
  alt=""
  blockProps={
    Object {
      "entity": Object {
        "getData": [Function],
      },
      "entityType": Object {},
    }
  }
  src="http://www.example.com/example.png"
>
  <a
    className="Tooltip__link EmbedBlock__link"
    href="http://www.example.com/"
    rel="noopener noreferrer"
    target="_blank"
    title="http://www.example.com/"
  >
    Test title
  </a>
  <button
    className="button button-secondary no Tooltip__button"
  >
    Delete
  </button>
</MediaBlock>
`;
